src/index.ts

```diff
@@ -10,9 +13,26 @@ import {
   RouteSegmentString,
   RouteSegmentParam
 } from './domain';
-import sortBy = require('lodash/sortBy');
-import uniq = require('lodash/uniq');
-import lowerFirst = require('lodash/lowerFirst');
+
+// FIXME(gabro): super-hack because I'm lazy and I don't want to pass this parameter around
+let _models: Array<Model>;
+
+function isNewtype(tpe: Tpe): boolean {
+  const model = _models.find(m => m.name === tpe.name);
+  return !!(model && 'isValueClass' in model && model.isValueClass);
+}
+
+const genericCombinator = (tpe: Tpe, prefix: string): gen.CustomCombinator => {
+  const type = gen.identifier(`${prefix}${tpe.name}`);
+  const staticArgs = tpe.args!.map(arg => gen.printStatic(getType(arg, false, prefix))).join(', ');
```

Comment:
It'd be nice to have the info about `args` encoded in the `Tpe` type (instead of casting with `!`). Anyway unrelated to this PR I guess

Reply:
yeah, I've discussed this with @gcanti today and I have some ideas.